l1b_lambda/level1b/handlers/level1b.py

```diff
@@ -0,0 +1,180 @@
+        rclone_config_path,
+        instrument_dir,
+        f"S3:{instrument_bucket}",
+    ))
+
+    return Instrument(
+        f"{instrument_dir}/calibration_data/calibration_data.toml",
+    )
+
+
+def lambda_handler(event: Event, context: Context):
+    output_bucket = get_env_or_raise("L1B_BUCKET")
+    instrument_bucket = get_env_or_raise("INSTRUMENT_BUCKET")
+    region = os.environ.get('AWS_REGION', "eu-north-1")
+    s3 = pa.fs.S3FileSystem(region=region)
+
+    try:
+        bucket, object = parse_event_message(event)
+        object_path = f"{bucket}/{object}"
+    except InvalidMessage:
+        return {
+            'statusCode': HTTPStatus.NO_CONTENT,
+            'headers': {'Content-Type': 'application/json'},
+            'body': json.dumps({
+                'message': 'Failed to parse event, nothing to do.'
+            })
+        }
+
+    if not object.endswith(".parquet"):
+        return {
+            'statusCode': HTTPStatus.NO_CONTENT,
+            'headers': {'Content-Type': 'application/json'},
+            'body': json.dumps({
+                'message': f'{object} is not a parquet file, nothing to do.'
+            })
+        }
+
+    with TemporaryDirectory(
+        "_instrument",
+        "/tmp",
+    ) as instrument_path:
+        rclone_config_path = get_rclone_config_path(
+            boto3.client("ssm"),
+            get_env_or_raise("RCLONE_CONFIG_SSM_NAME")
+        )
+        instrument = get_instrument(
+            instrument_path,
+            instrument_bucket,
+            rclone_config_path,
+        )
+
+        ccd_data = rpf.read_ccd_data(object_path, filesystem=s3)
+        ccd_items = rpf.dataframe_to_ccd_items(
+            ccd_data,
+            remove_empty=False,
+            remove_errors=False,
+            remove_warnings=False,
+        )
+
+        for ccd in ccd_items:
+            if ccd["IMAGE"] is None:
+                image_calibrated = None
+                errors = None
+            else:
+                (
+                    _,
+                    _,
+                    _,
+                    _,
+                    _,
+                    image_calibrated,
+                    _,
+                    errors,
+                ) = L1_calibrate(ccd, instrument)
+            ccd["ImageCalibrated"] = image_calibrated
+            ccd["CalibrationErrors"] = errors
+
+    calibrated = DataFrame.from_records(
+        ccd_items,
+        columns=["EXP Date", "ImageCalibrated", "CalibrationErrors"],
+    )
+    l1b_data = concat([
+        ccd_data,
+        calibrated,
+    ], axis=1).set_index("EXPDate").sort_index()
+    l1b_data.drop(["ImageData", "Errors", "Warnings"], axis=1, inplace=True)
+    l1b_data = l1b_data[l1b_data.ImageCalibrated != None]  # noqa: E711
+
+    out_table = pa.Table.from_pandas(l1b_data)
+    pq.write_table(
+        out_table,
+        f"{output_bucket}/{object_path}",
+        filesystem=s3,
+        version='2.6',
+    )
```

Comment:
It could be useful to wrap this part in a `try-except` that re-raises an error containing the file it tried to process.

Reply:
Fixed next push.